### ISSUE TITLE
[Dev] Wrote the Importer Module for Elementor.

### ISSUE THREAD

**Issue by redlagoon** [richtabor/MerlinWP#102]
@capuderg @richtabor 

In response to  #96

I've written the Elementor Importer module, tested it to hell and beyond, added error-checking and even wrote-up on the inner-workings.

Unfortunately, time / knowledge doesn't allow me to now integrate it fully within Merlin. Can you guys take a look and add it to steps?

https://gist.github.com/redlagoon/9c4bac3a87dd5bd3760bbe85beabae06

Here are some (important) notes:

https://gist.github.com/redlagoon/bd62f2899ad092b7265a599ddbd90b80

**Additionally, if you guys can give me a quick run-down of how to integrate the step, I'll do it myself, or at least try to. I'm currently in a big fog, the class is huge and it's hard to understand things for me.**

**Comment by Basilakis** [comment]
@capuderg @richtabor Any chance this can be merged? Or it can be a module?

**Comment by richtabor** [comment]
I agree with @capuderg. It's not on the priority list and I also don't use Elementor in any of my own themes.

**Comment by richtabor** [comment]
@redlagoon 

As far as we're concerned — we're building Merlin WP to suite our shops' particular needs. I've open sourced Merlin WP for other folks to chime in, offer suggestions, time, pull requests — the whole lot. And while I do absolutely appreciate your feedback, I'm under no obligation to act on any part of it. 

It does not matter if 99% of WordPress theme developers use a particular plugin or page builder, if contributors on this project are not — then it's highly unlikely an integration will happen. That's simply the nature of open source projects.: if there's a need, someone will work on it, submit it to the project and everyone will benefit. Otherwise, it doesn't happen.

Simply put, it's not something we'll use — we're not going to do it ourselves. 

On that not, you're 100% welcome to submit a PR with your suggested importer(s) in place, we'll review — and if it's all good — we'll gladly add support for it. There have already been a number of folks who've done just that and now Merlin WP is a much more capable wizard.

**Comment by redlagoon** [comment]
An update on this -

It turns out, you don't need to do this at all, so my preliminary research was in vain, but a good way to learn.

The way Elementor stores its page data, as well as settings is inside the post meta, which also gets imported by the WXR importer. On top of that, it stores its skeleton inside the page, as pure HTML.

That means Elementor is compatible. I'm still searching for issues as of now, I'll post here.

Damn.

**Comment by contempoinc** [comment]
@redlagoon So, in your tests all the pages created with Elementor work properly with the standard content XML export?
